mail_tracking_mass_mailing/models/mail_tracking_email.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+
+
+class MailTrackingEmail(models.Model):
+    _inherit = "mail.tracking.email"
+
+    mass_mailing_id = fields.Many2one(
+        string="Mass mailing", comodel_name='mail.mass_mailing', readonly=True)
+    mail_stats_id = fields.Many2one(
+        string="Mail statistics", comodel_name='mail.mail.statistics',
+        readonly=True)
+    mail_id_int = fields.Integer(string="Mail ID", readonly=True)
```

Comment:
This field is needed because in some cases mail.mail is deleted when event is triggered. Odoo implements this in the same way in mail.mail.statistics